event_handling/include/essentials/NotifyTimer.hpp

```diff
@@ -0,0 +1,155 @@
+#pragma once
+
+#include "ITrigger.hpp"
+
+#include <chrono>
+#include <condition_variable>
+#include <iostream>
+#include <thread>
+#include <vector>
+
+namespace essentials
+{
+
+template <class NotificationClass>
+using t_notificationcallback = void (NotificationClass::*)();
+
+/**
+ * The NotifyTimer allows to have a callback-member function be called in a fixed
+ * interval. The calling is delayed after each start by the given delay.
+ * @tparam NotificationClass
+ */
+template <class NotificationClass>
+class NotifyTimer
+{
+public:
+    NotifyTimer(std::chrono::milliseconds msInterval, std::chrono::milliseconds _msDelayedStart, t_notificationcallback<NotificationClass> callback,
+            NotificationClass* obj);
+    ~NotifyTimer();
+    /**
+     * Sets the timer to run.
+     * @return True, if the timer will run after this call. False, otherwise.
+     */
+    bool start();
+    /**
+     * Sets the timer to stop.
+     * @return True, if the timer will anyways run after this call. False, otherwise.
+     */
+    bool stop();
+    /**
+     * Allows to get the running state of the notify timer.
+     * @return True, if the timer is set to run. False, otherwise.
+     */
+    bool isStarted();
+    std::chrono::milliseconds getDelayedStart() const;
+    std::chrono::milliseconds getInterval() const;
+
+private:
+    void runInternal();                        /** < The method executed by the notify timer thread */
+    bool _running;                             /** < Is always true except when the notify timer is shutting down. */
+    bool _started;                             /** < True, if the NotifyTimer is active. False, otherwise. */
+    bool _justStartedAgain;                    /** < True, if the NotifyTimer needs to delay the next call, because it was (re)started. False, otherwise. */
+    std::chrono::milliseconds _msInterval;     /** < The milliseconds between two calls to the callback. */
+    std::chrono::milliseconds _msDelayedStart; /** < The milliseconds between (re)starting the NotifyTimer and the callback call. */
+    std::thread* _runThread;
```

Comment:
while you are at it, please change this to unique_ptr

Reply:
done